src/webui/abstractwebapplication.cpp

```diff
@@ -307,10 +308,12 @@ void AbstractWebApplication::increaseFailedAttempts()
 
 bool AbstractWebApplication::isAuthNeeded()
 {
-    return (env_.clientAddress != QHostAddress::LocalHost
-            && env_.clientAddress != QHostAddress::LocalHostIPv6
-            && env_.clientAddress != QHostAddress("::ffff:127.0.0.1"))
-            || Preferences::instance()->isWebUiLocalAuthEnabled();
+    qDebug("Checking auth rules against client address %s", qPrintable(env_.clientAddress.toString()));
+    if (Utils::Net::isLANAddress(env_.clientAddress))
```

Comment:
BTW, if your router does not perform ingress filtering, you will be extremely vulnerable to address spoofing, because any LAN address will bypass authentication.